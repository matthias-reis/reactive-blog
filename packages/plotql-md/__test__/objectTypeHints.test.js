const hint = require('../objectTypeHints');
const fixture = require('./fixture');

describe('objectTypeHints()', () => {
  it('hints a string', () => {
    expect(hint('foo')).toEqual('String');
  });

  it('hints a boolean', () => {
    expect(hint(true)).toEqual('Boolean');
    expect(hint(false)).toEqual('Boolean');
  });

  it('hints an integer', () => {
    expect(hint(42)).toEqual('Int');
  });

  it('hints a float', () => {
    expect(hint(42.42)).toEqual('Float');
  });

  it('hints a function', () => {
    expect(hint(() => {})).toEqual('Function');
  });

  it('hints an object', () => {
    expect(hint({ a: 42 })).toEqual({ a: 'Int' });
  });

  it('hints an object with more than one item', () => {
    expect(hint({ a: 42, b: 'foo' })).toEqual({
      a: 'Int',
      b: 'String'
    });
  });

  it('hints deeply nested objects', () => {
    expect(hint({ a: 42, b: { one: 1, two: 2 } })).toEqual({
      a: 'Int',
      b: { one: 'Int', two: 'Int' }
    });
  });

  it('hints an array of arbitrary types', () => {
    expect(hint([1, 'two', true])).toEqual(['Int', 'String', 'Boolean']);
  });

  it('hints an array of equal scalar types', () => {
    expect(hint([1, 2, 3])).toEqual(['Int']);
  });

  it('hints an array of equal object types', () => {
    expect(hint([{ a: 1 }, { a: 2 }])).toEqual([{ a: 'Int' }]);
  });

  it('hints an array of inequal object types', () => {
    expect(hint([{ a: 1 }, { a: 2, b: 'three' }])).toEqual([
      { a: 'Int', b: 'String' }
    ]);
  });

  it('hints null', () => {
    expect(hint(null)).toEqual(null);
  });

  it('hints complex objects', () => {
    expect(hint(fixture)).toMatchSnapshot();
  });
});
