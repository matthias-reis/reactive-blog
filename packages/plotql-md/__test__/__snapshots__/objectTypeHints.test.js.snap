// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`objectTypeHints() hints complex objects 1`] = `
Array [
  Object {
    "abv": "Float",
    "attenuation_level": "Int",
    "boil_volume": Object {
      "unit": "String",
      "value": "Int",
    },
    "brewers_tips": "String",
    "contributed_by": "String",
    "description": "String",
    "ebc": "Int",
    "first_brewed": "String",
    "food_pairing": Array [
      "String",
    ],
    "ibu": "Int",
    "id": "Int",
    "image_url": "String",
    "ingredients": Object {
      "hops": Array [
        Object {
          "add": "String",
          "amount": Object {
            "unit": "String",
            "value": "Float",
          },
          "attribute": "String",
          "name": "String",
        },
      ],
      "malt": Array [
        Object {
          "amount": Object {
            "unit": "String",
            "value": "Float",
          },
          "name": "String",
        },
      ],
      "yeast": "String",
    },
    "method": Object {
      "fermentation": Object {
        "temp": Object {
          "unit": "String",
          "value": "Int",
        },
      },
      "mash_temp": Array [
        Object {
          "duration": "Int",
          "temp": Object {
            "unit": "String",
            "value": "Int",
          },
        },
      ],
      "twist": null,
    },
    "name": "String",
    "ph": "Float",
    "srm": "Int",
    "tagline": "String",
    "target_fg": "Int",
    "target_og": "Int",
    "volume": Object {
      "unit": "String",
      "value": "Int",
    },
  },
]
`;
